base/src/main/java/io/spine/validate/MessageValue.java

```diff
@@ -89,11 +98,20 @@ static MessageValue atTopLevel(Message message) {
      */
     Optional<FieldValue> valueOf(String fieldName) {
         FieldDescriptor field = descriptor.findFieldByName(fieldName);
-        if (field == null) {
-            return Optional.empty();
-        }
-        FieldValue fieldValue = valueOf(field);
-        return Optional.of(fieldValue);
+        return valueOfNullable(field);
+    }
+
+    /**
+     * Obtains the value of a populated {@code OneOf} field.
+     *
+     * @param oneOf
+     *         the {@code OneOf} descriptor
+     * @return a value of the populated field or {@code Optional.empty()} if the message
```

Comment:
It seems to me that if the message does not contain the field, which description was passed to the method, we should throw an invalid argument exception.